forms-flow-api/src/formsflow_api/services/tasks.py

```diff
@@ -64,3 +76,95 @@ def get_task_outcome_configuration(
             response = task_outcome_schema.dump(task_outcome)
             return response
         raise BusinessException(BusinessErrorCode.TASK_OUTCOME_NOT_FOUND)
+
+    def form_submission(self, form_id: str, data: dict):
+        """Create form submission in formio."""
+        current_app.logger.debug("Formio new submission started...")
+        submission_data = {"data": data["form_data"].get("data")}
+        form_io_token = self.formio.get_formio_access_token()
+        formio_response = self.formio.create_submission(
+            form_id, submission_data, form_io_token
+        )
+        current_app.logger.debug("Formio new submission completed...")
+        return formio_response
+
+    def update_application(self, application_id: int, data: dict, user: UserContext):
+        """Update application details."""
+        application = Application.find_by_id(application_id=application_id)
+        if application is None:
+            raise BusinessException(BusinessErrorCode.APPLICATION_ID_NOT_FOUND)
+        # This can be used later if workflow completion fails and we need to rollback
+        application_backup_data = {
+            "application_status": application.application_status,
+            "submission_id": application.submission_id,
+            "latest_form_id": application.latest_form_id,
+        }
+        data["modified_by"] = user.user_name
+        application.update(data)
+        application.commit()
```

Comment:
It might be better to move the application update logic into `ApplicationService` and call that function here or from the below function where this is being called to maintain a clear separation of concerns.